src/random-seed/random-seed.c

```diff
@@ -115,20 +115,22 @@ static CreditEntropy may_credit(int seed_fd) {
         return CREDIT_ENTROPY_NO_WAY;
 }
 
-static int random_seed_size(int seed_fd, size_t *ret_size) {
+static size_t random_seed_size(int seed_fd) {
+        size_t poolsize;
         struct stat st;
 
-        assert(ret_size);
         assert(seed_fd >= 0);
 
-        if (fstat(seed_fd, &st) < 0)
-                return log_error_errno(errno, "Failed to stat() seed file " RANDOM_SEED ": %m");
+        poolsize = random_pool_size();
+
+        if (fstat(seed_fd, &st) < 0) {
```

Comment:
nah, fstat() cannot fail. if it does that should be a complete failure.

if we open a file, then this pins the inode, and that also means fstat() will always work. If it doesn't we are in deep trouble and then should definitely fail complete (it might be because the fd might be wrong, or so, i.e. EBADF or so.

Please, let's not try to recover from errors that cannot really happen.